src/strings.rs

```diff
@@ -0,0 +1,122 @@
+use std::ops::Deref;
+use std::borrow::{ToOwned, Borrow};
+use std::str::from_utf8_unchecked;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct StackStr {
+    inner: [u8; 4],
+    len: usize,
```

Comment:
A thought, `usize` is the natural size but in this case it might make sense to use a smaller type, `u8` is sufficient.

Reply:
Yup, sounds v sensible. Went for the minimal implementation that worked for arbitrary `char`s, but indeed there is room for other short strings. Doing the box → stack check on clone is pretty smart and something we could definitely do!